File: mobile/ios/SuperBrainContacts/Sources/Models/PathNode.swift
import Foundation

/// Path node representing a step in the shortest path between contacts
public struct PathNode: Codable, Identifiable, Equatable {
    public let contact: Contact
    public let distance: Int
    public let connectionType: String?
    
    public var id: UUID {
        contact.id
    }
    
    public init(
        contact: Contact,
        distance: Int,
        connectionType: String? = nil
    ) {
        self.contact = contact
        self.distance = distance
        self.connectionType = connectionType
    }
    
    /// Human-readable connection description
    public var connectionDescription: String {
        if let type = connectionType {
            return "\(type) (distance: \(distance))"
        }
        return "Distance: \(distance)"
    }
}

File: mobile/android/superbrain-contacts/app/src/main/java/com/superbrain/contacts/models/PathNode.kt
package com.superbrain.contacts.models

import java.util.UUID

/**
 * Path node representing a step in the shortest path between contacts
 */
data class PathNode(
    val contact: Contact,
    val distance: Int,
    val connectionType: String? = null
) {
    val id: UUID
        get() = contact.id
    
    /**
     * Human-readable connection description
     */
    val connectionDescription: String
        get() = connectionType?.let { "$it (distance: $distance)" } ?: "Distance: $distance"
}
